### PR TITLE
D8CORE-2490: All new alerts

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Updates NPM and all the packages
- Updates to font-awesome fonts and icons
- Updates to the Alerts component.
- Close functionality for alert.

# Needed By (Date)
- Sept 3rd

# Urgency
- Hight

# Steps to Test

`test.twig`
```
<div class="su-stuff">
  <h2>Things and stuff.</h2>
  <p>More things and stuff.</p>

  {% include "@decanter/components/alert/alert.twig" %}

  {% set alert_label = "Success" %}
  {% set modifier_class = "su-alert--success su-alert--text-light" %}
  {% include "@decanter/components/alert/alert.twig" %}

  {% set alert_label = "Info" %}
  {% set modifier_class = "su-alert--info su-alert--text-light" %}
  {% include "@decanter/components/alert/alert.twig" %}

  {% set alert_label = "Error" %}
  {% set modifier_class = "su-alert--error su-alert--text-light" %}
  {% include "@decanter/components/alert/alert.twig" %}

  {% set alert_label = "Warning" %}
  {% set modifier_class = "su-alert--warning" %}
  {% include "@decanter/components/alert/alert.twig" %}

  <p>&nbsp;</p>
  <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
</div>
```

1. Set up your developer environment and use the above test template
2. Compile all the assets (npm run publish)
3. Review the code
4. Review the responsive breakpoints and the aesthetics compared to https://www.figma.com/file/Kmd4utmJFPRMVeCFEEBQhLtx/Decanter-Design-System?node-id=9316%3A22
5. Review preview build

# Affected Projects or Products
- D8CORE
- Everything that used alerts before

# Associated Issues and/or People
- D8CORE-2490
- Adapt

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by yvonnetangsu** [comment]
@sherakama Is this ready for initial review?

**Comment by sherakama** [comment]
Not final review but if you want to take an initial review please have at it.

**Comment by yvonnetangsu** [comment]
@sherakama I think it's looking great. I like your use of utility classes (like the text-light) since it will upgrade to v7 better, and the use of flex order. Will review more thoroughly when it's ready.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by sherakama** [comment]
@yvonnetangsu This is ready for your critique. Thanks.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

File: core/src/scss/components/alert/_alert.scss [hunk 1]
```diff
@@ -7,51 +7,150 @@
...
+  flex-grow: 1;
+  flex-basis: 100px;
+}
+
+.su-alert__heading {
+  @include margin(null null 1rem null);
+}
+
+.su-alert__text {
+  @include modular-spacing('margin', null null 0 null);
```
Comment (yvonnetangsu) [inline core/src/scss/components/alert/_alert.scss:115]: Looks like if there is no footer included, then `.su-alert__text` and `.su-alert__label` will need a `margin-bottom: 0;` for vertical center alignment.
Screen Shot 2020-08-31 at 11 58 47 PM

File: core/src/scss/components/alert/_alert.scss [hunk 2]
```diff
@@ -7,51 +7,150 @@
...
+// Alert Type Styles.
+// ------------------
+.su-alert__label {
+  height: 100%;
+  line-height: 100%;
+  display: inline-block;
+  font-size: 1.7rem;
+  text-transform: uppercase;
+  font-weight: $su-font-semi-bold;
+  margin-bottom: 1em;
```
Comment (yvonnetangsu) [inline core/src/scss/components/alert/_alert.scss:95]: Please see the screenshot in the other comment - `.su-alert__label` will need `margin-bottom: 0;` whether or not there is footer.

File: core/src/scss/components/alert/_alert.scss (hunk 1 above)
Comment (sherakama) [reply core/src/scss/components/alert/_alert.scss:115]: Ah yesh. Margins. I'll have a look at those.

File: core/src/scss/core/fonts/_fontawesome-fonts.scss [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @charset "UTF-8";
 
 @font-face {
-  font-family: 'Font Awesome 5 Free';
+  font-family: "Font Awesome 5 Free";
```
Comment (yvonnetangsu) [inline core/src/scss/core/fonts/_fontawesome-fonts.scss:4]: FYI the way we include FontAwesome here, we don't have the "solid" set available - the "Free" set includes the solids and some of the "regulars", and the solids are in a separate file. For example, if you'd like to include the filled in version of the dismiss icon, we'll need to include the solid set in FontAwesome as well.

File: core/src/scss/components/alert/_alert.scss [hunk 4]
```diff
@@ -7,51 +7,150 @@
...
 //
-// .su-alert--success  - Successful Transaction Message
-// .su-alert--warning  - Warning Message
-// .su-alert--error    - Error Message
-// .su-alert--info     - Informative Message
+// .su-alert--success su-alert--text-light    - Successful Transaction Message
+// .su-alert--warning                         - Warning Message
+// .su-alert--error su-alert--text-light      - Error Message
+// .su-alert--info su-alert--text-light       - Informative Message
+// .su-alert--text-light                      - Lighter color text for darker backgrounds
```
Comment (yvonnetangsu) [inline core/src/scss/components/alert/_alert.scss:14]: Are we including the alert variant with the big icon in this PR, or just the small icon version?

Screen Shot 2020-09-01 at 10 43 55 AM

Comment (yvonnetangsu) [reply core/src/scss/components/alert/_alert.scss:14]: FYI looks like there are different padding values in the container depending on whether the alert has a big Alert Header and an Alert Footer.

File: core/src/scss/core/fonts/_fontawesome-fonts.scss (hunk 3 above)
Comment (yvonnetangsu) [reply core/src/scss/core/fonts/_fontawesome-fonts.scss:4]: Screen Shot 2020-09-01 at 10 57 15 AM

This screenshot is from another project - we might need to add the solid ones to the asset folder for Decanter too if we use them.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

File: core/src/scss/components/alert/_alert.scss (hunk 4 above)
Comment (sherakama) [reply core/src/scss/components/alert/_alert.scss:14]: Yeah, just the one variant for this release. I hope to come back to the other variants later.

File: core/src/scss/core/fonts/_fontawesome-fonts.scss (hunk 3 above)
Comment (sherakama) [reply core/src/scss/core/fonts/_fontawesome-fonts.scss:4]: Thanks, I noticed the missing asset on our builds. Should be fixed now.

**Comment by sherakama** [comment]
@yvonnetangsu assets and spacing has been adjusted.

**Comment by yvonnetangsu** [comment]
@sherakama Thanks for the change - this is super close. A couple small things:

- Might need a max-width (width minus the width of the dismiss button) on the alert__body element
Screen Shot 2020-09-01 at 12 35 52 PM
- Looks like we could use a larger padding when there are no alert header and alert footer - maybe a decorator class?  (screenshot showing current state - looks a little slim when there are no header and footer) 
Screen Shot 2020-09-01 at 12 36 30 PM

**Comment by sherakama** [comment]
> * Looks like we could use a larger padding when there are no alert header and alert footer - maybe a decorator class?  (screenshot showing current state - looks a little slim when there are no header and footer)

Yeah, I think one or two of the other variants will help in this case. The big icons or larger text options might be a better fit. I don't mind the slim notice overall.

**Comment by sherakama** [comment]
> Might need a max-width (something like width minus the width of the dismiss button or other approach) on the alert__body element

I just made sure to add some margin between the two on the XS screen size. That is the only size that gets that layout.

**Comment by sherakama** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr725-xhtj2ajwmgtxmmg3kkjavvvillxg0i0w.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5f47d954b750fa8d7c843d91

**Comment by yvonnetangsu** [comment]
> > * Looks like we could use a larger padding when there are no alert header and alert footer - maybe a decorator class?  (screenshot showing current state - looks a little slim when there are no header and footer)
> 
> Yeah, I think one or two of the other variants will help in this case. The big icons or larger text options might be a better fit. I don't mind the slim notice overall.

Yeah, I don't mind it being slim either.

**Comment by sherakama** [comment]
Winna winna.

**Pull request merged**